### ISSUE TITLE
Unable to suppress "Value of '{a}' may be overwritten in IE 8 and earlier." w/ -W034

### ISSUE THREAD

**Issue by jnewman** [jshint/jshint#2283]
I run JSHint on compiled CoffeeScript and I ran into a corner case where I had to use node:true for unintuitive reasons:
```coffee
foo = 
  try
    throw ''
  catch
    try
      throw ''
    catch
      42
```
compiles to:
```js
var foo;

foo = (function() {
  try {
    throw '';
  } catch (_error) {
    try {
      throw '';
    } catch (_error) {
      return 42;
    }
  }
})();
```
which warns that "Value of '_error' may be overwritten in IE 8 and earlier."

I'd expected to be abel to suppress this like other warnings that I'm not interested by `/* jshint -W034 */`, but that didn't work. 

As a workaround I added `/* jshint node:true */` and false around the block. Mostly just confusing that this wasn't suppressable by W034 or shadow.

**Comment by lukeapage** [comment]
it's because the warning number if `W002`

```
/* jshint -W002 */

var foo;

foo = (function() {
  try {
    throw '';
  } catch (_error) {
    try {
      throw '';
    } catch (_error) {
      return 42;
    }
  }
})();
```

**Comment by jnewman** [comment]
Thanks for the heads up. For posterity, I should have looked here: https://github.com/jshint/jshint/blob/06b57649414e44a9df6a46907f3ac93513e44221/src/messages.js#L79

**Comment by metagrapher** [comment]
I just ran into this issue on a try{ } catch (error) { } statement...
I just want to say that while I know there are like 0.0003% of developers supporting IE8 in December of 2022, this is kind of absurd. It's also absurd that it would trigger on only one of the 4 try/catch statements I have in this code.
I'm unsure if I have found some kind of a bug that is unrelated to this, or if jshint is just being absurdly verbose in its warnings. The idea that in 2022 I would have to turn off IE8 warnings, especially after setting my node version to 11, is just wild to me.

Anyhow, just commenting to mention that I have seen this in 2022.... If you're still supporting IE8... I'm so sorry.
![image](https://user-images.githubusercontent.com/59613197/206900435-d094b26f-a050-41c6-8051-747f3256c78d.png)
